incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSessionFactory.java

```diff
@@ -526,124 +840,68 @@ private void doMqttReset(
-            long affinity,
-            String16FW clientId,
-            String16FW clientIdMigrate,
-            String16FW sessionIdentifier)
+            long affinity)
         {
-            assert state == 0;
-
-            this.initialId = supplyInitialId.applyAsLong(routedId);
-            this.replyId = supplyReplyId.applyAsLong(initialId);
+            reconnectAttempt = 0;
+            willFetchers.values().forEach(f -> f.cleanup(traceId, authorization));
+            willFetchers.clear();
 
             state = MqttKafkaState.openingInitial(state);
 
-            kafka = newKafkaStream(this::onKafkaMessage, originId, routedId, initialId, initialSeq, initialAck, initialMax,
-                traceId, authorization, affinity, delegate.sessionsTopic, clientId, clientIdMigrate, sessionIdentifier);
-        }
-
-        private void doKafkaData(
-            long traceId,
-            long authorization,
-            long budgetId,
-            int reserved,
-            int flags,
-            OctetsFW payload,
-            Flyweight extension)
-        {
-
-            doData(kafka, originId, routedId, initialId, initialSeq, initialAck, initialMax,
-                traceId, authorization, budgetId, flags, reserved, payload, extension);
-
-            initialSeq += reserved;
-
-            assert initialSeq <= initialAck + initialMax;
-        }
-
-        private void doKafkaData(
-            long traceId,
-            long authorization,
-            long budgetId,
-            int reserved,
-            int flags,
-            MqttSessionStateFW sessionState,
-            Flyweight extension)
-        {
-            final DirectBuffer buffer = sessionState.buffer();
-            final int offset = sessionState.offset();
-            final int limit = sessionState.limit();
-            final int length = limit - offset;
-
-            doData(kafka, originId, routedId, initialId, initialSeq, initialAck, initialMax,
-                traceId, authorization, budgetId, flags, reserved, buffer, offset, length, extension);
-
-            initialSeq += reserved;
-
-            assert initialSeq <= initialAck + initialMax;
+            kafka = newWillStream(this::onWillMessage, originId, routedId, initialId, initialSeq, 0, 0,
+                traceId, authorization, affinity, sessionsTopic);
         }
 
         private void doKafkaEnd(
             long traceId,
-            long sequence,
             long authorization)
         {
             if (!MqttKafkaState.initialClosed(state))
             {
-                initialSeq = delegate.initialSeq;
-                initialAck = delegate.initialAck;
-                initialMax = delegate.initialMax;
                 state = MqttKafkaState.closeInitial(state);
 
-                doEnd(kafka, originId, routedId, initialId, initialSeq, initialAck, initialMax, traceId, authorization);
+                doEnd(kafka, originId, routedId, initialId, initialSeq, 0, 0, traceId, authorization);
```

Comment:
I think this was correct before the change, since we need to honor the values sent to use for `initialAck` and `initialMax` when sending the initial frames. There is no guarantee that it starts at `zero`.

Reply:
Actually git playing tricks here :). This is not a change, this a completely new code in the core0 will signal stream. I think we can assume that it's always 0, as there's no initial side for that. Or am I wrong?